homeassistant/components/huawei_lte/config_flow.py

```diff
@@ -184,6 +197,29 @@ def get_router_title(conn: Connection) -> str:
 
         return self.async_create_entry(title=title, data=user_input)
 
+    async def async_step_ssdp(self, discovery_info):
+        """Handle SSDP initiated config flow."""
+        # Try to distinguish from other non-LTE Huawei router devices
+        if (
+            discovery_info.get(ATTR_UPNP_DEVICE_TYPE)
+            != "urn:schemas-upnp-org:device:InternetGatewayDevice:1"
+            # e.g. "Mobile Wi-Fi"
+            or "mobile" not in discovery_info.get(ATTR_NAME, "").lower()
+        ):
+            return self.async_abort(reason="not_huawei_lte")
+
+        # https://github.com/PyCQA/pylint/issues/3167
+        url = self.context[CONF_URL] = url_normalize(  # pylint: disable=no-member
+            discovery_info.get(
+                ATTR_PRESENTATIONURL, f"http://{discovery_info[ATTR_HOST]}/"
+            )
+        )
+
+        if any(url == flow["context"][CONF_URL] for flow in self._async_in_progress()):
```

Comment:
There might be one in progress initiated by the user when it is discovered. That would trigger a `KeyError`. Use `get` instead:

```suggestion
        if any(url == flow["context"].get(CONF_URL) for flow in self._async_in_progress()):
```

Reply:
Should you also check if it has already been configured ?

Reply:
Ah, right, I initially implemented this by calling `async_step_import` at end here, and that took care of it, will fix.